### PR TITLE
docs: stop using --set for helm commands

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

##### Description

Fill in your description here.

---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/docs/Installation_with_Helm.md [hunk 1]
```diff
@@ -98,16 +98,25 @@ To install the chart, first add the `sumologic` private repo:
...
 ```
 
-Next you can run `helm upgrade --install` to install our chart.
-An example command with the minimum parameters is provided below.
-The following command will install the Sumo Logic chart with the release name
-`my-release` in the namespace your `kubectl` context is currently set to.
+Next you can prepare `values.yaml` with configuration.
+An example file with the minimum confiuration is provided below.
+It disables the `kube-prometheus-stack` sub-chart since
+we will be modifying the existing prometheus operator install.
```
Comment (sumo-drosiek) [inline deploy/docs/Installation_with_Helm.md:104]: its not

Comment (sumo-drosiek) [inline deploy/docs/Installation_with_Helm.md:104]:
```suggestion
```

File: deploy/docs/Installation_with_Helm.md [hunk 2]
```diff
@@ -268,9 +306,8 @@ If you wish to upgrade to a specific version, you can use the `--version` flag.
 helm upgrade --install my-release sumologic/sumologic -f values.yaml --version=2.0.0
 ```
 
-If you no longer have your `values.yaml` from the first installation or do not remember
-the options you added via `--set` you can run the following to see the values
-for the currently installed helm chart.
+__Note__ If you no longer have your `values.yaml` from the first installation
```
Comment (sumo-drosiek) [inline deploy/docs/Installation_with_Helm.md:309]:
```suggestion
__Note:__ If you no longer have your `values.yaml` from the first installation
```

File: deploy/docs/Kube-Prometheus.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ You can generate mixin configuration using `kubectl` or `docker`:
 
 The defaults assume you're deploying Sumo Logic collection via Helm and using few customizations.
 When deploying collection, disable the built-in Prometheus
-Operator either by using `--set kube-prometheus-stack.enabled=false` or editing `values.yml`:
+Operator either by editing `values.yml`:
```
Comment (sumo-drosiek) [inline deploy/docs/Kube-Prometheus.md:23]:
```suggestion
Operator by editing `values.yml`:
```

File: deploy/docs/SideBySidePrometheus.md [hunk 4]
```diff
@@ -306,7 +378,9 @@ If you wish to upgrade to a specific version, you can use the `--version` flag.
 helm upgrade --install my-release sumologic/sumologic -f values.yaml --version=1.0.0
 ```
 
-If you no longer have your `values.yaml` from the first installation or do not remember the options you added via `--set` you can run the following to see the values for the currently installed helm chart. For example, if the release is called `my-release` you can run the following.
+**Note** If you no longer have your `values.yaml` from the first installation
```
Comment (sumo-drosiek) [inline deploy/docs/SideBySidePrometheus.md:381]:
```suggestion
**Note:** If you no longer have your `values.yaml` from the first installation
```

File: deploy/docs/existingPrometheusDoc.md [hunk 5]
```diff
@@ -302,7 +329,9 @@ If you wish to upgrade to a specific version, you can use the `--version` flag.
 helm upgrade --install my-release sumologic/sumologic -f values.yaml --version=1.0.0
 ```
 
-If you no longer have your `values.yaml` from the first installation or do not remember the options you added via `--set` you can run the following to see the values for the currently installed helm chart. For example, if the release is called `my-release` you can run the following.
+__Note__ If you no longer have your `values.yaml` from the first installation
```
Comment (sumo-drosiek) [inline deploy/docs/existingPrometheusDoc.md:332]:
```suggestion
__Note:__ If you no longer have your `values.yaml` from the first installation
```

File: deploy/docs/standAlonePrometheus.md [hunk 6]
```diff
@@ -204,7 +239,9 @@ If you wish to upgrade to a specific version, you can use the `--version` flag.
 helm upgrade --install my-release sumologic/sumologic -f values.yaml --version=1.0.0
 ```
 
-If you no longer have your `values.yaml` from the first installation or do not remember the options you added via `--set` you can run the following to see the values for the currently installed Helm chart. For example, if the release is called `my-release` you can run the following.
+__Note__ If you no longer have your `values.yaml` from the first installation
```
Comment (sumo-drosiek) [inline deploy/docs/standAlonePrometheus.md:242]:
```suggestion
__Note:__ If you no longer have your `values.yaml` from the first installation
```

**Comment by swiatekm-sumo** [comment]
Should we encourage users to put their credentials in a separate file, for example `values-credentials.yaml`? It's a good practice to keep the credentials apart from the rest of the configuration. WDYT @aiwaniuk ?

**Comment by aiwaniuk** [comment]
thanks!

**Comment by sumo-drosiek** [comment]
@swiatekm-sumo I think we should rewrite huge part of the documentation as another PR, so we do not repeat 4 times how helm and configuration works.

I would make some document which describes how the configuration files are working, that the keys should be merged and also that every change require update

File: deploy/docs/Best_Practices.md [hunk 7]
```diff
@@ -1429,17 +1429,7 @@ In order to do that `nodeSelector` has to be used. By default node selectors can
...
-```bash
-helm upgrade --install my-release sumologic/sumologic \
-  --set sumologic.accessId="<SUMO_ACCESS_ID>" \
-  --set sumologic.accessKey="<SUMO_ACCESS_KEY>" \
-  --set sumologic.clusterName="<MY_CLUSTER_NAME>" \
-  --set fluent-bit.nodeSelector."kubernetes\.io/os"=linux 
-```
-
-or by modyfing above settings in values.yaml
+Node selector can be changed via additional parameter in `values.yaml`, below example for fluent-bit node selector:
```
Comment (swiatekm-sumo) [inline deploy/docs/Best_Practices.md:1432]:
```suggestion
Node selector can be changed via additional parameter in `values.yaml`, see an example for Fluent-Bit below:
```

File: deploy/docs/Non_Helm_Installation.md [hunk 8]
```diff
@@ -85,19 +85,29 @@ The following parameter is optional, but we recommend setting it.
...
+
+```yaml
+sumologic:
+  accessId: ${ACCESS_ID}
+  accessKey: ${ACCESS_KEY}
+  clusterName: ${CLUSTER_NAME}
+```
+
+Now you will need to generate the YAML to apply to your cluster.
+The following command contains the minimum parameters that can generate the YAML to setup Sumo Logic's Kubernetes collection.
```
Comment (swiatekm-sumo) [inline deploy/docs/Non_Helm_Installation.md:98]: Is this line necessary? We already said it was the minimal configuration above.

Comment (sumo-drosiek) [reply deploy/docs/Non_Helm_Installation.md:98]: I understand by minimal parameters that we refers to `helm` command. Imho it's not necessary to mention that. I can remove it

Comment (swiatekm-sumo) [reply deploy/docs/Non_Helm_Installation.md:98]: I think it's kind of confusing then, and we don't lose anything by removing it.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**